### ISSUE TITLE
How to make inference right

### ISSUE THREAD

**Issue by Podidiving** [PyTorchLightning/pytorch-lightning#2308]
Hello everyone. I'm new to pytorch-lightning, but already excited with this framework. It's very convenient to train my models using lightning. Now my usecase is: I have trained my model and want to do inference on my test data and get results (for example, in csv format). I'd like to do my inference pytorch-lightning-way. What is the best practice to do it?   

### Before asking:   
1. search the issues.   
2. search the docs.    

#### What's your environment?

 - OS: [Linux]
 - Packaging [pip]
 - Version [0.8.1]

**Comment by versatran01** [comment]
My understanding is that you just load the module and call freeze() on it and use it as any nn.Module

**Comment by rohitgr7** [comment]
You don't need to freeze(). Just run with `torch.no_grad()` or set `torch.set_grad_enabled(False)`.

**Comment by Podidiving** [comment]
I think, one of the coolest features of lightning is that your don’t need to specify your device (devices)
If you using LightningModule as plain nn.Module you should transfer your model and batches on devices manually, am I right? 
So, I have my trained model. I’d like to make inference on test data. I can define test_step and aggregate results in on_test_epoch_end, but I cannot run Trainer without train stage. Can I get around this somehow?

**Comment by rohitgr7** [comment]
`test_step` and `test_epoch_end` are called with `trainer.test()`. AFAIK, these methods are used just to evaluate your test dataset and not return any logits. If you want to do just this evaluation, you don't have to do the transfer of model or batches. But I think if you want to get the logits from the model you need to do it manually just like a vanilla PyTorch model.

**Comment by Podidiving** [comment]
Ok, thanks, didn’t know about `test` method! I think, if you want to make some kind of submission data, you can return logits from `test_step` method in dict object, and then aggregate and save them in `test_epoch_end`

**Comment by rohitgr7** [comment]
yeah, can do that too. Nice thought. :v:
